### PR TITLE
Modernize docs without making changes to its content

### PR DESCRIPTION
**Pull request by consideRatio**
- docs: update Makefile with devenv and linkcheck
- docs: update conf.py to not clutter as much
- docs: prepare for but don't switch to pydata theme yet
- docs: add linkcheck test
- docs: update links for the new linkcheck test
- Update .gitignore

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -1,32 +1,19 @@
...
-author = u'Yuvi Panda'
+# -- Project information -----------------------------------------------------
 
-# The short X.Y version
-version = u''
-# The full version, including alpha/beta/rc tags
-release = u'1.0'
+project = "Jupyter Server Proxy"
+copyright = f"{datetime.date.today().year}, Project Jupyter Contributors"
+author = "Project Jupyter Contributors"
```
Comment (consideRatio) [inline docs/conf.py:16]: @yuvipanda do you think this the right call of me to make, to update this like that?

You and Ryan are listed as authors of this package via a package.json that is read by setup.py as well still.

```json
  "author": {
    "name": "Ryan Lovett & Yuvi Panda",
    "email": "rylo@berkeley.edu"
  },
```

Comment (yuvipanda) [reply docs/conf.py:16]: Fine by me here. Should ping ryan too for the other bit :)

Comment (consideRatio) [reply docs/conf.py:16]: :+1: I'm not making that change!

**Comment by yuvipanda** [comment]
Thanks, @consideRatio

**Pull request merged**